nix/wire-server.nix

```diff
@@ -462,9 +462,35 @@ let
   allLocalPackagesBom = lib.buildBom allLocalPackages {
     includeBuildtimeDependencies = true;
   };
+
+  haskellPackages = hPkgs localModsEnableAll;
+  haskellPackagesUnoptimizedNoDocs = hPkgs localModsOnlyTests;
+
+  toplevel-derivations =
+    let
+      mk = pkg:
+        import ./pkg-info.nix {
+          inherit pkg;
+          inherit (pkgs) lib hostPlatform writeText;
+        };
+      out = import ./all-toplevel-derivations.nix {
+        inherit (pkgs) lib;
+        fn = mk;
+        recursionDepth = 2;
```

Comment:
Is there a special reason to choose `2` here? In this case we should document it.

Reply:
yes, after three it doesn't fit into memory.

Reply:
Does it still contain everything Emil needs to see?

Reply:
This :arrow_up: is actually my last main concern. Once we got this answered, I'll approve this PR.

Reply:
yes; there's only a couple of bootstrapping things missing; I explicitly put in package sets that we need.

Reply:
Cool :sunglasses:

Reply:
I mean cool is debatable. Nix not being able to work with the equivalent of 3 MB of JSON is concerning